### ISSUE TITLE
ECS task definition - image section missing uri part

### ISSUE THREAD

**Issue by jaydlawrence** [aws/copilot-cli#5261]
Hi.

Yesterday we did a small code change and pushed to our staging environment (using github actions which calls aws copilit svc deploy).

We are in `us-east-1` and copilot-cli version `v1.29.1`

We had a number of failed attempts to deploy the service.
Today we dug into it further and we found an issue with how the new task definitions looked in the ECS task definition JSON.

This is an example of what they used to look like:

```json
"image": "00000000.dkr.ecr.us-east-1.amazonaws.com/application-name/service-name:3b0b301",
```

But with these new ones, they look like this:

```json
"image": ":3f04320",
```

We have also tried to deploy from our local machines with:

```bash
copilot deploy --name service-name --env env-name --app application
```
As well as with the `--force` flag.

```bash
copilot deploy --name service-name --env env-name --app application --force
```

But we are getting the same issue.

We have other services that use stock images that don't have this issue, so my guess it that it has something to do with uploading the image to ecr or getting the image path once that has been done.

Is this likely to be something that has changed up-steam in CloudFormation that copilot hasn't been updated to handle?

**Comment by bvtujo** [comment]
Hmm, it looks like you're running into a problem which we introduced in 1.28.0 and should have fixed in 1.29.1. 

When you upgrade an app with >1.28.0 (which you should be prompted to do when necessary) , then initialize a new workload or environment using an *earlier* version of Copilot, it would accidentally delete all the ECR repos in the app. 

#5140 has more detail; do any of those problems look like what you're experiencing? There are workarounds there for getting affected apps back into a usable state.

**Comment by jaydlawrence** [comment]
We have tried the latest version of copilot locally, as well as the one in the mainline branch.
The mainline version generates Task Definitions with the image containing only the digest: `"image": "@sha..." and it won't even get to pushing task definition as CloudFormation kicks it back with:

```
The following resource(s) failed to update: [TaskDefinition].

...

Resource handler returned message: "Invalid request provided: Create TaskDefinition: Container.image repository should not be null or empty.
...

```

We were able to see what the task definition was that it was pushing up by adding some print lines to the copilot-cli code and compiling it and then trying to deploy the service with it.
We are not Go developers so we were mostly stumbling around in the dark.

![image](https://github.com/aws/copilot-cli/assets/1242060/74f01ad6-2e13-4123-b5a5-983d8f87769a)

![image](https://github.com/aws/copilot-cli/assets/1242060/1c2f30bb-ba3c-4955-9f4a-7f416b81672b)


The images have been pushing to ECR.
We have been able to sort of work around it by running it and having it push the image and then crash out with the above message and then swapping out the `build` section of our manifest with a `location` section pointing to the latest version of that image in ECR.
That is working, but we can't rely on that for deploying long-term or to our production environment.

But what it does tell us is that the image is reaching ECR and it seems to be the step that retrieves the URI and tag that is not getting the correct information.

For some extra information, we usually run this using github actions.

```
      name: Setup AWS Copilot
      uses: softprops/setup-aws-copilot@v1
```

It first broke there and we confirmed it and have been testing it by running it locally with various versions of `copilit-cli`.

If you have any suggestions for what we can do as far as debugging steps, I would be happy to help get to the bottom of it.
